azurelinuxagent/common/event.py

```diff
@@ -393,52 +529,106 @@ def _clean_up_message(message):
-            if param.name not in param_names or set_values_for_agent:
-                # If set_values_for_agent, we disregard any values already set for an existing default property and
-                # replaces it with a latest entry.
-                param_names[param.name] = param.value
+        for event_file in event_files:
+            try:
+                match = EVENT_FILE_REGEX.search(event_file)
+                if match is None:
+                    continue
 
-        parameters = DataContractList(TelemetryEventParam)
-        for name, value in param_names.items():
-            parameters.append(TelemetryEventParam(name, value))
+                event_file_path = os.path.join(event_directory_full_path, event_file)
 
-        return parameters
+                try:
+                    logger.verbose("Processing event file: {0}", event_file_path)
+
+                    with open(event_file_path, "rb") as fd:
+                        event_data = fd.read().decode("utf-8")
+
+                    event = parse_event(event_data)
+
+                    is_legacy_event = match.group('agent_event') is None
+
+                    if is_legacy_event:
+                        # We'll use the file creation time for the event's timestamp
+                        event_file_creation_time_epoch = os.path.getmtime(event_file_path)
+                        event_file_creation_time = datetime.fromtimestamp(event_file_creation_time_epoch)
+
+                        if event.is_extension_event():
+                            EventLogger._trim_extension_event_parameters(event)
+                            self._add_common_event_parameters(event, event_file_creation_time)
+                        else:
+                            self._update_old_event_schema(event, event_file_creation_time)
+
+                    event_list.events.append(event)
+                finally:
+                    os.remove(event_file_path)
+            except Exception as e:
+                logger.warn("Failed to process event file {0}: {1}", event_file, ustr(e))
+
+        return event_list
+
+    def _update_old_event_schema(self, event, event_creation_time):
+        # Ensure that if an agent event is missing a field from the schema defined since 2.2.47, the missing fields
+        # will be appended, ensuring the event schema is complete before the event is reported.
+        new_event = TelemetryEvent()
+        new_event.parameters = []
+        self._add_common_event_parameters(new_event, event_creation_time)
+
+        event_params = dict([(param.name, param.value) for param in event.parameters])
+        new_event_params = dict([(param.name, param.value) for param in new_event.parameters])
+
+        missing_params = set(new_event_params.keys()).difference(set(event_params.keys()))
+        params_to_add = []
+        for param_name in missing_params:
+            params_to_add.append(TelemetryEventParam(param_name, new_event_params[param_name]))
```

Comment:
Didn't we discuss not adding the (potentially wrong) value for missing telemetry fields? We can safely populate `OpcodeName` as it's a run-time value, but not others like `ContainerId` or `GAVersion`. This was a bug in #1757.

Reply:
Yes, that is the comment I posted at line 619. Looking forward for your PR fixing this (as we discussed) :)

Reply:
After our discussion, my understanding was that you weren't going to keep this logic in. Ok, will patch in a future PR.